lib/ansible/plugins/filter/core.py

```diff
@@ -102,6 +102,8 @@ def to_bool(a, strategy='yaml'):
             if a:
                 return True
     except TypeError as e:
+        if not strict:
+            return False
```

Comment:
I would also  warn

Reply:
It's not. The integer would be truthy instead of falsy.

Reply:
I don't think there is a way to restore the behavior now.